api/src/modules/projects/projects.service.ts

```diff
@@ -1,9 +1,17 @@
+import { FetchSpecification } from 'nestjs-base-service';
 import { Injectable } from '@nestjs/common';
 import { AppBaseService } from '@api/utils/app-base.service';
 import { Project } from '@shared/entities/projects.entity';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository, SelectQueryBuilder } from 'typeorm';
-import { FetchSpecification } from 'nestjs-base-service';
+
+interface ExtendedFetchSpecification extends FetchSpecification {
```

Comment:
For the maps, and for the consumers simplicity, I specified the filters (which match the project entity's properties) in the filters property of FetchSpecification, which is typed as Record<string, any>. Not ideal, we might refactor that approach

However if we go with extending the FetchSpecification, I think:
1- we should create an interface implementing the additional properties we need
2- we should pass it to the project contract's query specification so the consumer can infer the type, like

query: contract.type<FetchSpecification & WhateverNewProps>()

And then we can do the same in the extendFindAll

Reply:
Ok, after reviewing all the PR, I understand what you did here.

I think we could do it "simpler", let's talk later!